### DISCUSSION TITLE
Mermaid diagrams no showing in Github Pages

### DISCUSSION THREAD

**Discussion by b00nw33** [community/community#65040] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

I noticed that Mermaid diagrams display properly in markdown(.md) files, but they show up as their source code in Github Pages generated from these .md files (presumably by Jekyll). Is there a way to fix this?

**Comment by arafajs** [comment]
Yes, there is a way to fix this. GitHub Pages uses Jekyll to render Markdown files. By default, Jekyll does not support Mermaid diagrams. However, you can enable Mermaid support by following these steps:

1. In your repository, create a file called `_config.yml`.
2. In the `_config.yml` file, add the following line:
3. `mermaid: true`
4. Save the _config.yml file.

**Reply by b00nw33** [reply]
Thank you @arafajs for helping. I tried the minimalistic _config.yml file as shown below (for implementing Mermaid and Math input), but still could not get it to work.

```
theme: minima

mermaid: true

use_math: true
```

**Comment by antonheitz** [comment]
Hi All, 

just got it to work. All you need is a little extra code at the end of the html, either via `_include` or directly in the `default.html` in `_layout`. All you need is the official code from Mermaid on this page:

https://mermaid.js.org/intro/ (scroll down to the part on how to include it via script)

Currently this is:

```html
<script type="module">
    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
    mermaid.initialize({ startOnLoad: true });
</script>
```
Now just add in a small hack to the start of the script. Jekyll will give the wrapper of the mermaid diagrams the class `language-mermaid` while the script searches for `mermaid`. This will add the `mermaid` class to every element with the `language-mermaid` class before initializing the mermaid script:

```html
<script type="module">
    Array.from(document.getElementsByClassName("language-mermaid")).forEach(element => {
      element.classList.add("mermaid");
    });
    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
    mermaid.initialize({ startOnLoad: true });
</script>
```

See it in action here: https://antonheitz.github.io/xBRL-Forge/

Hope this helps!

**Reply by cben** [reply]
Nice.  If we're querying the document, we can avoid even loading the mermaid JS on pages that don't need it: https://stackoverflow.com/a/79671294/239657
